src/server/pageRendering.js

```diff
@@ -13,6 +13,34 @@ import reducers from '../shared/reducers';
 import { loadState } from './loadState';
 import { indexHtml } from './readIndexHtml';
 import App from '../shared/app';
+import fs from 'fs';
+import path from 'path';
+
+var IndexHtmlSingleton = function () {
+	var indexHtmlInstance;
+
+	function createInstance() {
+		const FILE_PATH = path.resolve('theme/assets/index.html');
+		fs.readFile(FILE_PATH, 'utf8', (err, data) => {
+			if (err) {
+				indexHtmlInstance = '';
+				winston.error('Fail to read file', FILE_PATH, err);
+			} else {
+				indexHtmlInstance = data;
+			}
+			return indexHtmlInstance;
+		});
+	}
+
+	return {
+		getInstance: function () {
+			if (!indexHtmlInstance) {
+				indexHtmlInstance = createInstance();
```

Comment:
add await to indexHtmlInstanc.getInstance function so if it null then it will wait until the instance is not loaded else it will again null for the same request but for other, it may have value.. to test this.. you can put a pause there.. and you will see it will bypass due to sync

Reply:
there is more perfect way with redux state?

Reply:
i sent new PR

Reply:
here is the commit - https://github.com/Cezerin2/cezerin2-store/pull/19/commits/5fd7b53c1e0a7d0cd8ebb1c1d496c3e1123f1e44